src/monstergenerator.cpp

```diff
@@ -422,13 +427,83 @@ void MonsterGenerator::finalize_mtypes()
             }
         }
         mon.weakpoints.finalize();
+
+        // check lastly to make extra fake monsters.
+        if( has_gen_dormant ) {
```

Comment:
Instead of checking all flags while they are being added to the set, just check the set down here, i.e. `mon.has_flag( mon_flag_GEN_DORMANT )`

Reply:
brainfart, you are correct. fixed.